File: python-pr/52/python/code-quality-readability-maintainability/text_analyzer_09.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def analyze_text(text):
    words = text.split()
    word_count = len(words)
    char_count = len(text)
    sentence_count = text.count('.') + text.count('!') + text.count('?')
    avg_word_length = sum(len(word) for word in words) / len(words) if words else 0
    return {'words': word_count, 'chars': char_count, 'sentences': sentence_count, 'avg_word_len': avg_word_length}
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/text_analyzer_09.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function analyzeText(text) {
    let words = text.split(' ');
    let wordCount = words.length;
    let charCount = text.length;
    let sentenceCount = (text.match(/[.!?]/g) || []).length;
    let avgWordLength = words.reduce((sum, word) => sum + word.length, 0) / words.length;
    return {words: wordCount, chars: charCount, sentences: sentenceCount, avgWordLen: avgWordLength};
}
// {/fact}